### ISSUE TITLE
package-info.java:29-43: The behaviours of the classes of...

### ISSUE THREAD

**Issue by 0pdd** [yegor256/cactoos#1242]
The puzzle `1184-56133cf3` from #1184 has to be resolved: 

https://github.com/yegor256/cactoos/blob/95f5a04afd70eb6cd0383c910ee9198b9ffd3119/src/main/java/org/cactoos/collection/package-info.java#L29-L43

The puzzle was created by Victor Noël on 02-Nov-19. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

**Comment by 0crat** [comment]
@paulodamaso[/z](https://www.0crat.com/u/paulodamaso) please, pay attention to this issue

**Comment by 0crat** [comment]
Job [#1242](https://github.com/yegor256/cactoos/issues/1242) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
The job [#1242](https://github.com/yegor256/cactoos/issues/1242) assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/C63314D6Z/c9dd903c-dbaa-4fd6-a48e-dd057ddc44f8); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by victornoel** [comment]
@paulodamaso can we have some guidance here? The todo explicitly asks about your opinion :)

**Comment by victornoel** [comment]
@0crat wait for ARC

**Comment by 0crat** [comment]
> @0crat wait for ARC ([here](https://github.com/yegor256/cactoos/issues/1242#issuecomment-557902372))

@victornoel The impediment for [#1242](https://github.com/yegor256/cactoos/issues/1242) was registered successfully by @victornoel[/z](https://www.0crat.com/u/victornoel)

**Comment by paulodamaso** [comment]
@victornoel I think that now that we have `Immutable`, we don't need to make `CollectionOf` immutable anymore.

**Comment by victornoel** [comment]
@paulodamaso the question is not so much about making `CollectionOf` immutable (because it is not actually), but about having it wrapping a `Scalar` (same with `ListOf` by the way) and be forced to use `Sticky` in the other decorators (e.g., `Filtered`) to be able to have them behave as expected.

I think the source of this problem, is always the same thing that I talked about in other comments:
- we have decorators for `Iterable` that really behave as decorators by augmenting the behaviour of `Iterable`
- we have decorators for `Collection` that are actually not decorator, but datastructure that contains the result of the class behaviour
  - for example, filtered behave as a collection of the wrapped collection filtered the FIRST time it is used.

I think we should rely only on `Iterable` decorators to augment the behaviour of all collections (because they all implement `Iterable`) and simply:
1. Remove `CollectionOf` (see #1252)
2. Have `ListOf` and `SetOf` behave as purely datastructure, like `LinkedList`, or `ArrayList, with nicer constructors taking an `Iterable`.
3. Use the following pattern if we need to iterate over a list using as a an example :
```java
new Filtered(s -> s.contains("a"), new ListOf("aa", "ab", "bc")).forEach(System.out::println);
```
4. Use the following pattern if we need a new list for example filtered content:
```java
List myList = new ListOf(new Filtered(s -> s.contains("a"), new ListOf("aa", "ab", "bc")));
```
5. Remove all `Collection`, `List` and `Set` decorators that becomes useless now (things like `Shuffled` or `Reversed` could still make sense and should not be decorators of `Iterable` or `Iterator` since they suppose the storage of state to be implemented, which is contrary to those interfaces contract.

**Comment by paulodamaso** [comment]
@victornoel I think that finally I understood the problem. I really see no point in having a lot of `List` and `Collection` decorators, since all of our behavior can sit nicely in the `Iterable` decorations. But we still have to trim some edges here:
1. Already put #1252 in scope and start removing these bad guys;
1. I'm thinking about all these data structures and persistent data stuff, and in my opinion I think we should back up from these ideas a little for now. I think that cactoos main objective is to provide extended behaviors to an abstract data structure (hence using the `Iterable` interface for everyone makes sense) in a elegant (declarative) way. So I don't think that we should bother with the storage of state of these objects, but just with the behavior of these decorations. 
1. I disagree with this. I still want to have an declarative way of iterating a collection in cactoos. We currently have `And` but as stated in https://github.com/yegor256/cactoos/issues/1226 it is clearly having more than onre responsibility, so fixing that one should solve the iteration problem;
1. At the end, all of decorations should look like this I think;
1. I disagree in not implementing these behaviors as decorators. I think that decoration is one of the core ideas of cactoos and elegant objects doctrine; again, I think that the main focus of cactoos, for now, is to provide these behaviors in an elegant and declarative form.

WDYT?

**Comment by victornoel** [comment]
@paulodamaso cool :)

About:
2. Yes, I agree, why not maybe in the future, but if we should proritize, I think we should just remove from scope every ticket pertaining to data structures and collections, and maybe later who knows we can get them back in.
3. Ok, I get your point, it's good with me yes!
5. I'm not clear what would be their behaviour in the end, something like this?
```java
public final class Filtered extends ListEnvelope {
  public Filtered(Predicate filter, List origin) {
    super(new ListOf(new Filtered(filter, origin)));
  }
}
```
If yes, then I think it's coherent with the rest so I wouldn't be complaining about it ;)

**Comment by paulodamaso** [comment]
@victornoel Just clearing 5. , maybe I got it wrong. I was thinking in removing all decorations for `List` and `Set`, so we should have only `ListOf` and `SetOf` in these packages. But as you stated correctly, some of our decorators doesn't make sense for `Iterable` or `Iterator`, but just for `List` (we don't have any decorator for `Set`). Also, looking through all these classes made me think that the reason they exist maybe is because is too verbose to use `ListOf` every time we have an Iterable and want to decorate it and make it a list. So your proposal is ok, we don't need to remove them all, but just wrap our Iterables with them correctly.

So, summarizing:
1. removing collections, starting on https://github.com/yegor256/cactoos/issues/1252
1. let data structures implementation for a future release
1. iterate using `ForEach`, to be implemented in https://github.com/yegor256/cactoos/issues/1226
1. See 5.
1. Change all `List` decorators that have its counterpart in `Iterable` package to wrap the `Iterable` one, and keep the ones that does not have siblings in `Iterable` package implemented directly in `List` package

Anything to add?

**Comment by victornoel** [comment]
@paulodamaso great, that seem cool.

So basically the present issue could start by making `ListOf` to be an actual implementation of `List` that can be used normally without this awkward  `Scalar` thing in it (after the PR for #1185 is merged because until then, the awkward `Scalar` was in the envelope)?

Once this is done we can then remove all those `Sticky` that are currently needed everywhere (as mentioned in the `@todo` of this issue) and have a clear simple behaviour (because honestly, currently it is hell, every time you change a little things, every tests break, see for example: https://github.com/yegor256/cactoos/pull/1255#issuecomment-562932482).

Actually I think `list.Sticky` could be deleted because it wouldn't make any sense anymore if `ListOf` is actually just a `List` (while `iterable.Sticky` stills makes sense for example).

And then do the same for `SetOf`.

**Comment by victornoel** [comment]
@paulodamaso WDYT?

**Comment by paulodamaso** [comment]
@victornoel I think we're good, let's follow your suggestions

**Comment by victornoel** [comment]
@0crat wait for #1255 to be merged first.

**Comment by 0crat** [comment]
> @0crat wait for #1255 to be merged first. ([here](https://github.com/yegor256/cactoos/issues/1242#issuecomment-581032332))

@victornoel Job [#1242](https://github.com/yegor256/cactoos/issues/1242) is already on hold

**Comment by victornoel** [comment]
@paulodamaso I did the work in #1242, don't hesitate to double check already to see if I incorrectly approached this for some points discussed above :)

**Comment by 0pdd** [comment]
The puzzle `1184-56133cf3` has disappeared from the source code, that's why I closed this issue.

**Comment by 0pdd** [comment]
@0pdd 14 puzzles [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1303](https://github.com/yegor256/cactoos/issues/1303), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1305](https://github.com/yegor256/cactoos/issues/1305) are still not solved.

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @victornoel[/z](https://www.0crat.com/u/victornoel), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Comment by 0crat** [comment]
Order was finished, quality is "good": +35 point(s) just awarded to @victornoel[/z](https://www.0crat.com/u/victornoel)

**Comment by 0pdd** [comment]
@0pdd 13 puzzles [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1303](https://github.com/yegor256/cactoos/issues/1303), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1305](https://github.com/yegor256/cactoos/issues/1305) are still not solved; solved: [#1294](https://github.com/yegor256/cactoos/issues/1294).

**Comment by 0pdd** [comment]
@0pdd 10 puzzles [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1305](https://github.com/yegor256/cactoos/issues/1305) are still not solved; solved: [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1303](https://github.com/yegor256/cactoos/issues/1303).

**Comment by 0pdd** [comment]
@0pdd 10 puzzles [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1305](https://github.com/yegor256/cactoos/issues/1305), [#1320](https://github.com/yegor256/cactoos/issues/1320) are still not solved; solved: [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1303](https://github.com/yegor256/cactoos/issues/1303).

**Comment by 0pdd** [comment]
@0pdd 10 puzzles [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1305](https://github.com/yegor256/cactoos/issues/1305), [#1320](https://github.com/yegor256/cactoos/issues/1320), [#1325](https://github.com/yegor256/cactoos/issues/1325) are still not solved; solved: [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1303](https://github.com/yegor256/cactoos/issues/1303).

**Comment by 0pdd** [comment]
@0pdd 9 puzzles [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1305](https://github.com/yegor256/cactoos/issues/1305), [#1320](https://github.com/yegor256/cactoos/issues/1320), [#1325](https://github.com/yegor256/cactoos/issues/1325), [#1326](https://github.com/yegor256/cactoos/issues/1326) are still not solved; solved: [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1303](https://github.com/yegor256/cactoos/issues/1303), [#1304](https://github.com/yegor256/cactoos/issues/1304).

**Comment by 0pdd** [comment]
@0pdd 8 puzzles [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1305](https://github.com/yegor256/cactoos/issues/1305), [#1320](https://github.com/yegor256/cactoos/issues/1320), [#1325](https://github.com/yegor256/cactoos/issues/1325) are still not solved; solved: [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1303](https://github.com/yegor256/cactoos/issues/1303), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1326](https://github.com/yegor256/cactoos/issues/1326).

**Comment by 0pdd** [comment]
@0pdd 7 puzzles [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1305](https://github.com/yegor256/cactoos/issues/1305), [#1320](https://github.com/yegor256/cactoos/issues/1320) are still not solved; solved: [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1303](https://github.com/yegor256/cactoos/issues/1303), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1325](https://github.com/yegor256/cactoos/issues/1325), [#1326](https://github.com/yegor256/cactoos/issues/1326).

**Comment by 0pdd** [comment]
@0pdd 4 puzzles [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1320](https://github.com/yegor256/cactoos/issues/1320) are still not solved; solved: [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1303](https://github.com/yegor256/cactoos/issues/1303), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1305](https://github.com/yegor256/cactoos/issues/1305), [#1325](https://github.com/yegor256/cactoos/issues/1325), [#1326](https://github.com/yegor256/cactoos/issues/1326).

**Comment by 0pdd** [comment]
@0pdd 4 puzzles [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1320](https://github.com/yegor256/cactoos/issues/1320), [#1337](https://github.com/yegor256/cactoos/issues/1337) are still not solved; solved: [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1303](https://github.com/yegor256/cactoos/issues/1303), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1305](https://github.com/yegor256/cactoos/issues/1305), [#1325](https://github.com/yegor256/cactoos/issues/1325), [#1326](https://github.com/yegor256/cactoos/issues/1326).

**Comment by 0pdd** [comment]
@0pdd 2 puzzles [#1320](https://github.com/yegor256/cactoos/issues/1320), [#1337](https://github.com/yegor256/cactoos/issues/1337) are still not solved; solved: [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1303](https://github.com/yegor256/cactoos/issues/1303), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1305](https://github.com/yegor256/cactoos/issues/1305), [#1325](https://github.com/yegor256/cactoos/issues/1325), [#1326](https://github.com/yegor256/cactoos/issues/1326).

**Comment by 0pdd** [comment]
@0pdd 2 puzzles [#1337](https://github.com/yegor256/cactoos/issues/1337), [#1349](https://github.com/yegor256/cactoos/issues/1349) are still not solved; solved: [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1303](https://github.com/yegor256/cactoos/issues/1303), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1305](https://github.com/yegor256/cactoos/issues/1305), [#1320](https://github.com/yegor256/cactoos/issues/1320), [#1325](https://github.com/yegor256/cactoos/issues/1325), [#1326](https://github.com/yegor256/cactoos/issues/1326).

**Comment by 0pdd** [comment]
@0pdd the puzzle [#1349](https://github.com/yegor256/cactoos/issues/1349) is still not solved; solved: [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1303](https://github.com/yegor256/cactoos/issues/1303), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1305](https://github.com/yegor256/cactoos/issues/1305), [#1320](https://github.com/yegor256/cactoos/issues/1320), [#1325](https://github.com/yegor256/cactoos/issues/1325), [#1326](https://github.com/yegor256/cactoos/issues/1326), [#1337](https://github.com/yegor256/cactoos/issues/1337).

**Comment by 0pdd** [comment]
@0pdd all 19 puzzles are solved here: [#1292](https://github.com/yegor256/cactoos/issues/1292), [#1293](https://github.com/yegor256/cactoos/issues/1293), [#1294](https://github.com/yegor256/cactoos/issues/1294), [#1295](https://github.com/yegor256/cactoos/issues/1295), [#1296](https://github.com/yegor256/cactoos/issues/1296), [#1297](https://github.com/yegor256/cactoos/issues/1297), [#1298](https://github.com/yegor256/cactoos/issues/1298), [#1299](https://github.com/yegor256/cactoos/issues/1299), [#1300](https://github.com/yegor256/cactoos/issues/1300), [#1301](https://github.com/yegor256/cactoos/issues/1301), [#1302](https://github.com/yegor256/cactoos/issues/1302), [#1303](https://github.com/yegor256/cactoos/issues/1303), [#1304](https://github.com/yegor256/cactoos/issues/1304), [#1305](https://github.com/yegor256/cactoos/issues/1305), [#1320](https://github.com/yegor256/cactoos/issues/1320), [#1325](https://github.com/yegor256/cactoos/issues/1325), [#1326](https://github.com/yegor256/cactoos/issues/1326), [#1337](https://github.com/yegor256/cactoos/issues/1337), [#1349](https://github.com/yegor256/cactoos/issues/1349).
